bedrock/src/transaction/contracts/safe_owner.rs

```diff
@@ -0,0 +1,97 @@
+//! This module introduces the contract interface for the Safe contract.
+//!
+//! Explicitly this only allows management of the Safe Smart Account. Executing transactions with the Safe Smart Account
+//! is done via the `SafeSmartAccount` module.
+
+use alloy::{
+    primitives::{address, Address, Bytes},
+    sol,
+    sol_types::SolCall,
+};
+
+use crate::{
+    primitives::PrimitiveError,
+    smart_account::{
+        InstructionFlag, Is4337Encodable, NonceKeyV1, TransactionTypeId, UserOperation,
+    },
+};
+
+const SENTINEL_ADDRESS: Address =
+    address!("0x0000000000000000000000000000000000000001");
+
+sol! {
+    ///Owner Manager Interface for the Safe
+    ///
+    /// Reference: <https://github.com/safe-global/safe-smart-account/blob/v1.4.1/contracts/base/OwnerManager.sol>
+    #[derive(serde::Serialize)]
+    #[sol(rename_all = "camelcase")]
+    interface IOwnerManager {
+        function swapOwner(address prev_owner, address old_owner, address new_owner) public;
+    }
+}
+
+/// Represents a Safe owner swap transaction for key rotation.
+pub struct SafeOwner {
+    /// The inner call data for the ERC-20 `transferCall` function.
+    call_data: Vec<u8>,
+    /// The address of the Safe Smart Account.
+    wallet_address: Address,
+}
+
+impl SafeOwner {
+    /// Creates a new `SafeOwner` transaction for swapping Safe owners.
+    ///
+    /// # Arguments
+    /// - `wallet_address`: The address of the Safe Smart Account
+    /// - `old_owner`: The current owner to be replaced
+    /// - `new_owner`: The new owner to replace the old owner
+    #[must_use]
+    pub fn new(
+        wallet_address: Address,
+        old_owner: Address,
+        new_owner: Address,
+    ) -> Self {
+        Self {
+            call_data: IOwnerManager::swapOwnerCall {
+                prev_owner: SENTINEL_ADDRESS,
```

Comment:
is this always the case? I'm not familiar with this logic in the contract 

What if a user rotates for the 2nd/3rd time